apps/omg_watcher_rpc/lib/web/validators/deposit_constraints.ex

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherRPC.Web.Validator.DepositConstraints do
+  @moduledoc """
+  Validates query parameters for `deposit.all`
+  """
+
+  alias OMG.WatcherRPC.Web.Validator.Helpers
+
+  @doc """
+  Validates possible query constraints, stops on first error.
+  """
+  @spec parse(%{binary() => any()}) :: {:ok, Keyword.t()} | {:error, any()}
+  def parse(params) do
+    constraints = [
+      {"limit", [pos_integer: true, lesser: 1000, optional: true], :limit},
```

Comment:
what does lesser do? are we setting a max cap you're allowed to fetch?

Reply:
Correct. This is defined in `validators/base.ex`. I did not give much thought to the number itself but this is consistent with `transaction.all` and `block.all`.

Reply:
Is there a test?

Reply:
Also, max parameters needs to be quite strict if we don't want to overflow watcher info. So make a reasonable amount that can be queried. We can increase it gradually if we see the need.

Reply:
No tests for `lesser` in base_test.ex so I wrote them (a2dcd7e) as this was an easy one. 

I think defining a maximum limit parameter for our endpoints is out of scope for this PR and best left for more dedicated consideration.

Reply:
Not saying 1000 is the right number. Just keeping things consistent